lib/traject/config/sirsi_config.rb

```diff
@@ -33,22 +33,51 @@
+to_field 'vern_title_related_search', extract_marc('505t:700fgklmnoprst:710dfgklmnoprst:711fgklnpst:730adfgklmnoprst:740anp:760st:762st:765st:767st:770st:772st:773st:774st:775st:776st:777st:780st:785st:786st:787st:796fgklmnoprst:797dfgklmnoprst:798fgklnpst:799adfgklmnoprst', alternate_script: :only)
+# Title Display Fields
+to_field 'title_245a_display', extract_marc('245a', alternate_script: false, trim_punctuation: true)
+to_field 'vern_title_245a_display', extract_marc('245a', alternate_script: :only, trim_punctuation: true)
+to_field 'title_245c_display', extract_marc('245c', alternate_script: false, trim_punctuation: true)
+to_field 'vern_title_245c_display', extract_marc('245c', alternate_script: :only, trim_punctuation: true)
+to_field 'title_display', extract_marc('245abdefghijklmnopqrstuvwxyz', alternate_script: false, trim_punctuation: true)
+to_field 'vern_title_display', extract_marc('245abdefghijklmnopqrstuvwxyz', alternate_script: :only, trim_punctuation: true)
+to_field 'title_full_display', extract_marc('245abcdefghijklmnopqrstuvwxyz', alternate_script: :false)
+to_field 'vern_title_full_display', extract_marc('245abcdefghijklmnopqrstuvwxyz', alternate_script: :only)
+to_field 'title_uniform_display', extract_marc(%w(130 240).map { |c| "#{c}abcdefghijklmnopqrstuvwxyz" }.join(':'), first: true, alternate_script: false)
 # # ? no longer will use title_uniform_display due to author-title searching needs ? 2010-11
-# title_uniform_display = custom, getAllAlphaSubfields(130:240, first)
-# vern_title_uniform_display = custom, getVernacular(130abcdefghijklmnopqrstuvwxyz:240abcdefghijklmnopqrstuvwxyz, first)
+# TODO: Remove looks like SearchWorks is not using, confirm relevancy changes
+to_field 'vern_title_uniform_display', extract_marc(%w(130 240).map { |c| "#{c}abcdefghijklmnopqrstuvwxyz" }.join(':'), first: true, alternate_script: :only)
 # # Title Sort Field
-# title_sort = custom, getSortTitle
+to_field 'title_sort' do |record, accumulator|
+  result = []
+  result << extract_sortable_title('130abcdefghijklmnopqrstuvwxyz', record)
+  result << extract_sortable_title('245abdefghijklmnopqrstuvwxyz', record)
+  accumulator << result.join(' ').strip
+end
+
+##
+# Originally cribbed from Traject::Marc21Semantics.marc_sortable_title, but by
+# using algorithm from StanfordIndexer#getSortTitle.
+def extract_sortable_title(fields, record)
+  java7_regex_punct = '!"#$%&\'()*+,-./:;<=>?@[\]^_`{|}~'
+  Traject::MarcExtractor.new(fields).collect_matching_lines(record) do |field, spec, extractor|
+    str = extractor.collect_subfields(field, spec).first
+    if str.nil?
+      # maybe an APPM archival record with only a 'k'
+      str = field['k']
+    end
+    if str.nil?
+      # still? All we can do is bail, I guess
+      return nil
+    end
+
+    non_filing = field.indicator2.to_i
+    str = str.slice(non_filing, str.length)
+    str = str.delete(java7_regex_punct).delete('$').strip
```

Comment:
I think the `.delete('$')` is now covered by the special java punctuation list (which isn't a regex, so maybe the variable should be renamed?)

Reply:
Good point, I've fixed it up now.